kedro-telemetry/kedro_telemetry/plugin.py

```diff
@@ -168,26 +168,23 @@ def before_command_run(
     ):
         """Hook implementation to send command run data to Heap"""
 
-        if not project_metadata:  # in package mode
-            return
+        project_path = project_metadata.project_path if project_metadata else None
 
-        self._consent = _check_for_telemetry_consent(project_metadata.project_path)
+        self._consent = _check_for_telemetry_consent(project_path)
         if not self._consent:
             self._opt_out_notification()
             return
 
         # get KedroCLI and its structure from actual project root
-        cli = KedroCLI(project_path=project_metadata.project_path)
+        cli = KedroCLI(project_path=project_path if project_path else Path.cwd())
```

Comment:
I guess my question is, does that `project_path` here actually matters? Where is it used as the `command_args` are coming from hook and `KedroCLI` which was created in `kedro` framework.

Reply:
I am also trying to understand why KedroCLI is dependent on a specific Kedro project. I believe the CLI structure should be the same for all projects. Do you think we can just use Path.cwd() here?

Reply:
I don't think the `project_path` matters but we should add a comment to explain this. I have questions about the `_recurse_cli` used by the function, does it takes care of the LazyCommand we added lately?

Reply:
> I have questions about the `_recurse_cli` used by the function, does it takes care of the LazyCommand we added lately?

I checked the [Lazy Loading PR](https://github.com/kedro-org/kedro/pull/3883), looks like `_recurse_cli()` func should work as previously. Also I tested current PR manually with latest kedro main branch on fresh virtual env, everything works well from the 1st command.

Reply:
I think Nok is right, the recursion through the CLI definitely structure triggers the loading of lazy the CLI command groups -
It uses `get_command()` :
https://github.com/kedro-org/kedro-plugins/blob/63a0c91f16b2149d083892788615e51b9a11edf1/kedro-telemetry/kedro_telemetry/masking.py#L44
Which if you look at the implementation of the `LazyGroup`, triggers the lazy load: 
https://github.com/kedro-org/kedro/blob/8bbbfb6c7257005bef841143690308f05d4b829b/kedro/framework/cli/utils.py#L504-L509

I believe we need to find a way to do the masking without loading relying on KedroCLI to not defeat the purpose of introducing lazy loading